test/Elastic.Apm.Tests/BasicAgentTests.cs

```diff
@@ -87,6 +87,33 @@ public async Task PayloadSentWithBearerToken()
 			authHeader.Parameter.Should().Be(secretToken);
 		}
 
+		[Fact]
+		public async Task PayloadSentWithProperUserAgent()
+		{
+			var isRequestFinished = new TaskCompletionSource<object>();
+
+			HttpHeaderValueCollection<ProductInfoHeaderValue> userAgentHeader = null;
+			var handler = new MockHttpMessageHandler((r, c) =>
+			{
+				userAgentHeader = r.Headers.UserAgent;
+				isRequestFinished.SetResult(null);
+				return Task.FromResult(new HttpResponseMessage(HttpStatusCode.OK));
+			});
+
+			var logger = ConsoleLogger.Instance;
+			var service = Service.GetDefaultService(new TestAgentConfigurationReader(logger), logger);
+			var payloadSender = new PayloadSenderV2(logger, new TestAgentConfigurationReader(logger),
+				service, new Api.System(), handler);
+
+			using (var agent = new ApmAgent(new TestAgentComponents(payloadSender: payloadSender)))
+			{
+				agent.PayloadSender.QueueTransaction(new Transaction(agent, "TestName", "TestType", new TestAgentConfigurationReader(logger)));
+			}
+
+			await isRequestFinished.Task;
+			userAgentHeader.Should().ContainSingle(x => x.Product.Name == "elasticapm-dotnet" && x.Product.Version == service.Agent.Version);
```

Comment:
I recommend to enforce the order and verify it in the test. Based on [RFC 2616](https://tools.ietf.org/html/rfc2616#section-14.43)
> By convention, the product tokens are listed in order of their significance for identifying the application.

Based on https://github.com/elastic/apm/issues/88#issuecomment-491192841 `elasticapm-$language/$agent_version` should be the first.

Reply:
fixed